samcli/hook_packages/terraform/hooks/prepare/enrich.py

```diff
@@ -0,0 +1,646 @@
+"""
+Terraform resource enrichment
+"""
+import logging
+import json
+import os
+from typing import Dict, List, Tuple
+from json.decoder import JSONDecodeError
+from samcli.hook_packages.terraform.lib.utils import (
+    CFN_CODE_PROPERTIES,
+    SAM_METADATA_RESOURCE_TYPE_ATTRIBUTE,
+    SAM_METADATA_DOCKER_TAG_ATTRIBUTE,
+    SAM_METADATA_DOCKER_FILE_ATTRIBUTE,
+    SAM_METADATA_DOCKER_BUILD_ARGS_ATTRIBUTE,
+    SAM_METADATA_RESOURCE_NAME_ATTRIBUTE,
+    _get_python_command_name,
+    get_sam_metadata_planned_resource_value_attribute,
+    _calculate_configuration_attribute_value_hash,
+    build_cfn_logical_id,
+)
+from samcli.hook_packages.terraform.hooks.prepare.types import SamMetadataResource
+from samcli.hook_packages.terraform.hooks.prepare.makefile import (
+    _generate_makefile_rule_for_lambda_resource,
+    _generate_makefile,
+)
+from samcli.hook_packages.terraform.hooks.prepare.resource_linking import _resolve_resource_attribute
+from samcli.hook_packages.terraform.hooks.prepare.exceptions import InvalidSamMetadataPropertiesException
+from samcli.lib.utils.resources import (
+    AWS_LAMBDA_FUNCTION as CFN_AWS_LAMBDA_FUNCTION,
+    AWS_LAMBDA_LAYERVERSION as CFN_AWS_LAMBDA_LAYER_VERSION,
+)
+from samcli.lib.utils.packagetype import ZIP, IMAGE
+from samcli.lib.hook.exceptions import PrepareHookException
+
+
+LOG = logging.getLogger(__name__)
+
+
+def _enrich_resources_and_generate_makefile(
```

Comment:
Some of these methods are no longer protected/private, we should remove the underscores from the ones that are imported else where